babel/numbers.py

```diff
@@ -20,46 +20,104 @@
 #  - http://www.unicode.org/reports/tr35/ (Appendix G.6)
 import re
 from datetime import date as date_, datetime as datetime_
+from itertools import chain
 
-from babel.core import default_locale, Locale, get_global
+from babel.core import default_locale, Locale, get_global, UnknownLocaleError
 from babel._compat import decimal
+from babel.localedata import locale_identifiers
 
 
 LC_NUMERIC = default_locale('LC_NUMERIC')
 
 
-def get_currency_name(currency, count=None, locale=LC_NUMERIC):
+def list_currency(locale=None):
+    """ Return a list of normalized currency codes.
```

Comment:
The docstring is lying in that this may return either a `dict_keys` object (when `locale` is passed in) or a `set`, but never a list.

Maybe always return a `set`?